### PR TITLE
Docker examples

### PR DESCRIPTION
**Pull request by fguisso**
Hello guys, I've been trying to script for a few months to automate my simnet environments and also update the example with LN docker-compose.

For now I have succeeded in building a Minimum Voting Wallet (MVW), see #12 , and interacting with `dcrd` and` dcrwallet` in a nice way. Just add a docker-compose up --build to get everything up and running.

I'm still going to improve some things, but I've opened PR so they can give me a light and tell if it's all by default since I'm not an expert docker.

What we have so far:
You can choose wallet_seed and wallet_pass from `docker-compose.yml` or using --build-args.
Thus, when docker creates the application, dcrwallet dockerfile uses two steps, one for creating the wallet, generating an address and saving that address to a volume like dcrd.conf, and the other for leaving the container clean only with what is needed. for the dcrwallet to run.
With this, `dcrd` runs with --miningaddr flag set for your wallet and you can generate blocks and init dcrwallet with ticketbuyer.

Next steps:
- fix the `dcrlnd` files to work with this MVW
- organize and update readme to find more UX tips to simplify example use.

I'm using this repos to refer in how MVW works: 
- https://github.com/matheusd/simnet-tmuxes
- https://github.com/fguisso/dcr-docker

### CODE REVIEW

**Comment by fguisso** [comment]
Draft for doc
**General workflow is the following:** 

 * Run `docker-compose up -d --build` to build/start services `dcrd`, `dcrwallet`, `dcrctl` and `dcrlnd` with name `Alice`.

Now we need to stop `dcrwallet` service and generate 100 blocks for the first coinbase to mature, because without coins, the MVW can't buy tickets, cause a crash in the service.

 * Stop the wallet service `docker-compose stop dcrwallet`
 * Generate 100 blocks `docker-compose run dcrctl generate 100`
 * Then restart wallet service again to automagic buy tickets `docker-compose start dcrwallet`

Send funds to `Alice`:

 * We need to create an address to `Alice` receive funds from MVW `dcrwallet`, then go inside `Alice`s container and use `dcrlncli` tool.
```bash
#Log into the "Alice" container:
`docker exec -it alice bash`

# Generate a new backward compatible nested p2sh address for Alice:
alice$ dcrlncli --n simnet newaddress p2pkh
```
We can keep logged in `Alice` container, just need to use another terminal tab to execute commands in docker-compose.

 * Send 1DCR from `dcrwallet` to `Alice` address `docker-compose run dcrctl --wallet sendfrom deafult <alice_address> 1`

 * Create `Bob`s container and send funds from `dcrwallet`:
```bash
#Create "Bob"s container
`docker-compose run -d --name bob dcrlnd`

#Log into the "Bob" container:
`docker exec -it bob bash`

# Generate a new backward compatible nested p2sh address for Bob:
bob$ dcrlncli --n simnet newaddress p2pkh
```
* Send 1DCR from `dcrwallet` to `Bob` address `docker-compose run dcrctl --wallet sendfrom deafult <bob_address> 1`

 * Include this transactions in the block
`docker-compose run dcrctl generate 6`

Check `Alice` and `Bob` balance:
```bash
alice$ dcrlncli -n simnet walletbalance

bob$ dcrlncli -n simnet walletbalance
```

**to be continue...**

**Comment by fguisso** [comment]
@matheusd and @jholdstock can you review please?
I stopped by simnet since we need MVW, but the upstream LND tutorial has a part about using `testnet`, wanted an idea as to whether it is better to create another compose just for testnet or continue the example with more dcrlnd nodes?
https://github.com/lightningnetwork/lnd/tree/master/docker#connect-to-faucet-lightning-node

File: docker/README.md [hunk 1]
```diff
@@ -1,100 +1,111 @@
...
+        + -------------- +        "dcrd" node which represents the Decred network, in a
+                                  real situation Alice and Bob will likely be
+                                  connected to different Decred nodes.
 ```
 
-**General workflow is the following:** 
+**General workflow is the following:**
+
+* Create a `dcrd` node running on a private `simnet`.
+* Create `Alice`, one of the `dcrlnd` nodes in our simulation network.
```
Comment (matheusd) [inline docker/README.md]: Needs to specify the MVW

**Comment by jholdstock** [comment]
I've run into some issues whilst testing this, going to debug with @fguisso on Matrix

File: docker/dcrlnd/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+# Expose dcrlnd ports (server, rpc).
+EXPOSE 9735 10009
+
+# Copy the binaries from the builder image.
+COPY --from=builder /go/bin/dcrlncli /bin/
+COPY --from=builder /go/bin/dcrlnd /bin/
+
+# Add bash.
+RUN apk add --no-cache \
```
Comment (jholdstock) [inline docker/dcrlnd/Dockerfile]: Moving the `RUN apk add` above the `COPY` statements will avoid re-running the bash installation every time dcrlncli or dcrlnd binaries change

Comment (jholdstock) [reply docker/dcrlnd/Dockerfile]: You resolved this one, but the change has not been made. I suggest installing `bash`, `git` and `make` all in one line, at the top of the file.

**Pull request merged**